composer/lib/dependabot/composer/file_parser.rb

```diff
@@ -50,7 +62,40 @@ def ecosystem
 
       sig { returns(Ecosystem::VersionManager) }
       def package_manager
-        PackageManager.new(composer_version)
+        raw_composer_version = env_versions[:composer] || composer_version
+        PackageManager.new(
+          raw_composer_version
+        )
+      end
+
+      sig { returns(T.nilable(Ecosystem::VersionManager)) }
```

Comment:
is it possible to return N/A in case this is also not defined instead of nil?

Reply:
This is a instance object, It will be hard to return `N/A` in case if we don't have version. If we want to do that we need to return T.any(String, Ecosystem::VersionManager), 2 different types which will make it more complicated.